conans/test/remote_checks_test.py

```diff
@@ -0,0 +1,138 @@
+
+        client.save({"conanfile.py": conanfile % "Server3!"})
+        client.run("create . Pkg/0.1@lasote/testing")
+        bump_time(40)
+        client.run("upload Pkg* -r=server3 --confirm --all")
+
+        # The remote defined is the first one that was used for upload
+        client.run("remote list_ref")
+        self.assertIn("Pkg/0.1@lasote/testing: server1", client.out)
+
+        client.run("remove * -f")
+        client.run("install Pkg/0.1@lasote/testing -r=server1")
+        self.assertIn("Pkg/0.1@lasote/testing: Server1!", client.out)
+        client.run("install Pkg/0.1@lasote/testing -r=server2")
+        self.assertIn("Pkg/0.1@lasote/testing: Server1!", client.out)
+        # Update
+        client.run("install Pkg/0.1@lasote/testing -r=server2 --update")
+        self.assertIn("Pkg/0.1@lasote/testing: Server2!", client.out)
+        client.run("remote list_ref")
+        self.assertIn("Pkg/0.1@lasote/testing: server2", client.out)
+        # Update
+        client.run("install Pkg/0.1@lasote/testing -r=server3 --update")
+        self.assertIn("Pkg/0.1@lasote/testing: Server3!", client.out)
+        client.run("remote list_ref")
+        self.assertIn("Pkg/0.1@lasote/testing: server3", client.out)
+
+    def test_binary_defines_remote(self):
+        servers = {"server1": TestServer(), "server2": TestServer(), "server3": TestServer()}
+        client = TestClient(servers=servers, users={"server1": [("lasote", "mypass")],
+                                                    "server2": [("lasote", "mypass")],
+                                                    "server3": [("lasote", "mypass")]})
+        conanfile = """from conans import ConanFile
+class Pkg(ConanFile):
+    pass"""
+        client.save({"conanfile.py": conanfile})
+        client.run("create . Pkg/0.1@lasote/testing")
+        client.run("upload Pkg* --all -r=server1 --confirm")
+        client.run("upload Pkg* --all -r=server2 --confirm")
+
+        # It takes the default remote
+        client.run("remove * -f")
+        client.run("remote list_ref")
+        self.assertNotIn("Pkg", client.out)
+
+        client.run("export . Pkg/0.1@lasote/testing")
+        client.run("install Pkg/0.1@lasote/testing")
+        self.assertIn("Downloading conan_package.tgz", client.out)
+        client.run("remote list_ref")
+        self.assertIn("Pkg/0.1@lasote/testing: server1", client.out)
+
+        # Explicit remote also defines the remote
+        client.run("remove * -f")
+        client.run("remote list_ref")
+        self.assertNotIn("Pkg", client.out)
+        client.run("export . Pkg/0.1@lasote/testing")
+        client.run("install Pkg/0.1@lasote/testing -r=server2")
+        self.assertIn("Downloading conan_package.tgz", client.out)
+        client.run("remote list_ref")
+        self.assertIn("Pkg/0.1@lasote/testing: server2", client.out)
+
+        # But order fails!!!
+        client.run("remove * -f")
+        client.run("remove * -f -r=server1")
```

Comment:
Why is this happening? Is it not iterating the remotes? is a simple install...

Reply:
No, it is not iterating the remotes. Current behavior for locally exported package recipe is:
- Check if -r=remote is defined
- Use default remote otherwise

Reply:
Feels like a bug, right? If it has no associated remote maybe it should iterate.